examples/wifi/expanded/requests_wifi_api_openskynetwork_private.py

```diff
@@ -19,135 +18,178 @@
 # JSON order: transponder, callsign, country
 # ACTIVE transpondes only, for multiple "c822af&icao24=cb3993&icao24=c63923"
-transponder = "7c6b2d"
+TRANSPONDER = "ad4f1c"
 
-# Initialize WiFi Pool (There can be only 1 pool & top of script)
-pool = socketpool.SocketPool(wifi.radio)
+# Get WiFi details, ensure these are setup in settings.toml
+ssid = os.getenv("CIRCUITPY_WIFI_SSID")
+password = os.getenv("CIRCUITPY_WIFI_PASSWORD")
+osnusername = os.getenv("OSN_USERNAME")  # Website Credentials
+osnpassword = os.getenv("OSN_PASSWORD")  # Website Credentials
 
-# Time between API refreshes
+# API Polling Rate
 # 900 = 15 mins, 1800 = 30 mins, 3600 = 1 hour
 # OpenSky-Networks IP bans for too many requests, check rate limit.
 # https://openskynetwork.github.io/opensky-api/rest.html#limitations
-sleep_time = 1800
+SLEEP_TIME = 1800
 
-# Get WiFi details, ensure these are setup in settings.toml
-ssid = os.getenv("CIRCUITPY_WIFI_SSID")
-password = os.getenv("CIRCUITPY_WIFI_PASSWORD")
-osnu = os.getenv("OSN_Username")
-osnp = os.getenv("OSN_Password")
-
-osn_cred = str(osnu) + ":" + str(osnp)
-bytes_to_encode = b" " + str(osn_cred) + " "
-base64_string = base64.encodebytes(bytes_to_encode)
-base64cred = repr(base64_string)[2:-1]
-
-Debug_Auth = False  # STREAMER WARNING this will show your credentials!
-if Debug_Auth:
-    osn_cred = str(osnu) + ":" + str(osnp)
-    bytes_to_encode = b" " + str(osn_cred) + " "
-    print(repr(bytes_to_encode))
-    base64_string = base64.encodebytes(bytes_to_encode)
-    print(repr(base64_string)[2:-1])
-    base64cred = repr(base64_string)[2:-1]
-    print("Decoded Bytes:", str(base64cred))
+# Set debug to True for full JSON response.
+# WARNING: makes credentials visible
+DEBUG = False
+
+# Initalize Wifi, Socket Pool, Request Session
+pool = adafruit_connection_manager.get_radio_socketpool(wifi.radio)
+ssl_context = adafruit_connection_manager.get_radio_ssl_context(wifi.radio)
+requests = adafruit_requests.Session(pool, ssl_context)
+
+# -- Base64 Conversion --
+OSN_CREDENTIALS = str(osnusername) + ":" + str(osnpassword)
+# base64 encode and strip appended \n from bytearray
+OSN_CREDENTIALS_B = binascii.b2a_base64(b"" + str(OSN_CREDENTIALS)).strip()
+BASE64_STRING = str(OSN_CREDENTIALS_B)  # bytearray
+SLICED_BASE64_STRING = BASE64_STRING[2:-1]  # slice bytearray head/tail
+
+if DEBUG:
+    print("Base64 ByteArray: ", BASE64_STRING)
+    print(f"Base64 Sliced String: {SLICED_BASE64_STRING}")
 
 # Requests URL - icao24 is their endpoint required for a transponder
 # example https://opensky-network.org/api/states/all?icao24=a808c5
-# OSN private requires your username:password to be base64 encoded
-osn_header = {"Authorization": "Basic " + str(base64cred)}
-OPENSKY_SOURCE = "https://opensky-network.org/api/states/all?" + "icao24=" + transponder
+# OSN private: requires your website username:password to be base64 encoded
+OPENSKY_HEADER = {"Authorization": "Basic " + str(SLICED_BASE64_STRING)}
```

Comment:
```suggestion
OPENSKY_HEADER = {"Authorization": "Basic " + BASE64_STRING}
```

Reply:
Very nice.  That makes things easier.  Didn't know how to use encode and decode.  Learning a lot here thank you!  

I'm not sure what's up with their hash checking.  Maybe an attempt at optimization, maybe lazy, or a bit of both.  Either way it's not the right way to do it, I'm tempted to try to contact them and request they check the full hash.  Odds are if you have the first 18 of the hash the rest is going to be correct but it's still not the right way to do it.